packages/flutter/lib/src/widgets/sensitive_content.dart

```diff
@@ -0,0 +1,311 @@
+      case ContentSensitivity.notSensitive:
+        _notSensitiveWigetCount++;
+      // ignore is safe because it protects this setting from tracking SensitiveContent
+      // widgets with an _unknown ContentSensitivity. _unknown is private to avoid
+      // developers using it as a SensitiveContent sensitivity level.
+      // ignore: no_default_cases
+      default:
+        throw FlutterError(
+          'Adding a SensitiveContent widget with ContentSensitivity $sensitivityLevel is unsupported by ContentSensitivitySetting',
+        );
+    }
+  }
+
+  /// Decreases the count of [SensitiveContent] widgets with [sensitivityLevel] set.
+  void removeWidgetWithContentSensitivity(ContentSensitivity sensitivityLevel) {
+    switch (sensitivityLevel) {
+      case ContentSensitivity.sensitive:
+        _sensitiveWidgetCount--;
+      case ContentSensitivity.autoSensitive:
+        _autoSensitiveWidgetCount--;
+      case ContentSensitivity.notSensitive:
+        _notSensitiveWigetCount--;
+      // ignore is safe because it protects this setting from tracking SensitiveContent
+      // widgets with an _unknown ContentSensitivity. _unknown is private to avoid
+      // developers using it as a SensitiveContent sensitivity level.
+      // ignore: no_default_cases
+      default:
+        throw FlutterError(
+          'Removing a SensitiveContent widget with ContentSensitivity $sensitivityLevel is unsupported by ContentSensitivitySetting',
+        );
+    }
+  }
+
+  /// Returns true if this class is currently tracking at least one [SensitiveContent] widget.
+  bool get hasWidgets =>
+      max(0, _sensitiveWidgetCount) +
+          max(0, _autoSensitiveWidgetCount) +
+          max(0, _notSensitiveWigetCount) >
+      0;
+
+  /// Returns the highest prioritized [ContentSensitivity] level of the [SensitiveContent] widgets
+  /// that this setting tracks.
+  ContentSensitivity? get contentSensitivityBasedOnWidgetCounts {
+    if (_sensitiveWidgetCount > 0) {
+      return ContentSensitivity.sensitive;
+    }
+    if (_autoSensitiveWidgetCount > 0) {
+      return ContentSensitivity.autoSensitive;
+    }
+    if (_notSensitiveWigetCount > 0) {
+      return ContentSensitivity.notSensitive;
+    }
+    return null;
+  }
+}
+
+/// Host of the current content sensitivity level for the widget tree that contains
+/// some number [SensitiveContent] widgets.
+@visibleForTesting
+class SensitiveContentHost {
+  SensitiveContentHost._();
+
+  bool? _contentSenstivityIsSupported;
+  late final ContentSensitivitySetting _contentSensitivitySetting = ContentSensitivitySetting();
+  ContentSensitivity? _fallbackContentSensitivitySetting;
+
+  final SensitiveContentService _sensitiveContentService = SensitiveContentService();
+
+  /// The [ContentSensitivity] level set for the entire widget tree calculated from
+  /// [_contentSensitivitySetting].
+  @visibleForTesting
+  ContentSensitivity? calculatedContentSensitivityLevel;
```

Comment:
Looking at the code, this variable seems to track what the framework thinks that Android has for the ContentSensitivity. It's not really calculated. Maybe change the name and the docs here?

Reply:
I actually discussed this name with Reid. I think it is calculated because kinda like you said, it's what the framework has calculated the setting to be. I think it's appropriate naming, but if you have strong feelings about changing it, let me know.

Reply:
Yes, naming is hard. if either of yall can some up with something better I am all for it. Apart from a suggestion for a better name I think the only thing we can do is explain the complexity in the dart doc which this variable has.

Reply:
Ok I see, yeah no strong feelings on the name, but I think we should explain in the docs. Maybe something like:

> The last known [ContentSensitivity] held by the platform for this Flutter view.